editors/vscode/server/src/linter.rs

```diff
@@ -228,9 +232,12 @@ impl IsolatedLintHandler {
         linter: &Linter,
         path: &Path,
         plugin: Plugin,
+        source_text: Option<String>,
     ) -> Option<(PathBuf, Vec<ErrorWithPosition>)> {
-        let source_text =
-            fs::read_to_string(path).unwrap_or_else(|_| panic!("Failed to read {path:?}"));
+        let source_text = source_text.unwrap_or_else(|| {
```

Comment:
I don't get this. What's the source of truth when it comes to reading source code in vscode? Is it from the disk or some other vscode API?

Reply:
It comes from the language server protocol, you can get the up-to-date source code from it, current problem is that we would get the outdated source code if the user doesn't trigger the save action. 
https://github.com/oxc-project/oxc/blob/023e6eabcd5121ff1a231b5e1fcf8e9b2cf607d4/editors/vscode/server/src/linter.rs#L232C8-L233C92

Reply:
For more details, you could refer https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentContentChangeEvent.